### PR TITLE
Makecode update

### PR DESCRIPTION
**Pull request by rakuy0**
A tiny bit of cleanup, but mostly changing makeCode to allow for a function to be passed in, where the new fragment will be linked to the function instead of being a standalone fragment (this is in support of a few switchcase updates that are hurdling down the pike).

And among those, this also introduces two new VaSets, EmucodeFunctions and CodeFragments to give better access to the results of other analyses.

### CODE REVIEW

File: vivisect/analysis/generic/emucode.py [hunk 1]
```diff
@@ -103,12 +101,11 @@ def analyze(vw):
 
     flist = vw.getFunctions()
 
-    tried = {}
-    vasetrows = []
+    tried = set()
```
Comment (rakuy0) [inline vivisect/analysis/generic/emucode.py:104]:
```
import time
import random

x = set()
y = dict()

nums = [random.randint(1, 1000000) for i in range(10000000)]
set_time = time.time()
for i in nums:
    if i in x:
        continue
    x.add(i)
set_len = time.time() - set_time


dict_time = time.time()
for i in nums:
    if y.get(i):
        continue
    y[i] = True
dict_len = time.time() - dict_time

print('set -- %f' % set_len)
print('dict -- %f' % dict_len)
```

Comment (rakuy0) [reply vivisect/analysis/generic/emucode.py:104]: Results (all numbers are in seconds):
1:
```
set -- 5.827403
dict -- 7.569130
```
2:
```
set -- 5.605092
dict -- 6.598526
```

3:
```
set -- 6.686243
dict -- 7.955484
```

File: vivisect/qt/funcgraph.py [hunk 2]
```diff
@@ -501,7 +501,7 @@ def _renderMemory(self):
 
             try:
                 addr = self.vw.parseExpression(expr)
-            except Exception, e:
+            except Exception as e:
```
Comment (rakuy0) [inline vivisect/qt/funcgraph.py:504]: This arose out of me trying to view the code fragments in funcgraph. That failed, but at least we got some cleanup out of it.

File: vivisect/__init__.py [hunk 3]
```diff
@@ -151,8 +151,12 @@ def __init__(self):
         self.addVaSet('DynamicBranches', (('va', VASET_ADDRESS), ('opcode', VASET_STRING), ('bflags', VASET_INTEGER)))
         self.addVaSet('SwitchCases', (('va', VASET_ADDRESS), ('setup_va', VASET_ADDRESS), ('Cases', VASET_INTEGER)))
         self.addVaSet('PointersFromFile', (('va', VASET_ADDRESS), ('target', VASET_ADDRESS), ('file', VASET_STRING), ('comment', VASET_STRING), ))
+<<<<<<< HEAD
         self.addVaSet('CodeFragments', (('va', VASET_ADDRESS), ('calls_from', VASET_COMPLEX)))
         self.addVaSet('EmucodeFunctions', (('va', VASET_ADDRESS),))
+=======
+        self.addVaSet('FuncWrappers', (('va', VASET_ADDRESS), ('wrapped_va', VASET_ADDRESS),))
+>>>>>>> master
```
Comment (williballenthin) [inline vivisect/__init__.py]: danger!

Comment (rakuy0) [reply vivisect/__init__.py]: Heh. Yea. i noticed that as soon as I threw this up for consideration. Should be fixed now.

Comment (atlas0fd00m) [reply vivisect/__init__.py]: nice catch!

File: vivisect/base.py [hunk 4]
```diff
@@ -740,16 +740,7 @@ def _cb_function(self, fva, fmeta):
...
-            fmod = vw.fmods.get(fmname)
-            try:
-                logger.debug('fmod: 0x%x  (%r)', fva, fmod)
-                fmod.analyzeFunction(vw, fva)
-            except Exception as e:
-                if vw.verbose:
-                    traceback.print_exc()
-                vw.verbprint("Function Analysis Exception for 0x%x   %s: %s" % (fva, fmod.__name__, e))
-                vw.setFunctionMeta(fva, "%s fail" % fmod.__name__, traceback.format_exc())
+        vw.analyzeFunction(fva)
```
Comment (atlas0fd00m) [inline vivisect/base.py:743]: :+1:

File: vivisect/__init__.py [hunk 5]
```diff
@@ -1353,6 +1377,7 @@ def makeFunction(self, va, meta=None, arch=envi.ARCH_DEFAULT):
         if meta is not None:
             for key, val in meta.items():
                 self.setFunctionMeta(realfva, key, val)
+        return realfva
```
Comment (atlas0fd00m) [inline vivisect/__init__.py:1380]: :+1:

File: vivisect/__init__.py [hunk 6]
```diff
@@ -1287,6 +1306,11 @@ def makeCode(self, va, arch=envi.ARCH_DEFAULT):
 
         logger.debug("makeCode(0x%x, 0x%x)", va, arch)
         calls_from = self.cfctx.addCodeFlow(va, arch=arch)
+        if fva is None:
```
Comment (atlas0fd00m) [inline vivisect/__init__.py:1309]: we should actually discuss a `getFunctions()` call that returns all the functions an opcode or codeblock belongs to.

File: vivisect/analysis/generic/emucode.py [hunk 7]
```diff
@@ -108,12 +107,11 @@ def analyze(vw):
...
```
Comment (atlas0fd00m) [inline vivisect/analysis/generic/emucode.py:110]: :+1:   performance is king!  nice job.

File: vivisect/qt/funcgraph.py (hunk 2 above)
Comment (atlas0fd00m) [inline vivisect/qt/funcgraph.py:504]: yay unit tests!

**Comment by atlas0fd00m** [comment]
tests are failing and branch is out of date.  
approved, as soon as tests succeed and the branch is updated ;)

File: vivisect/__init__.py (hunk 6 above)
Comment (rakuy0) [reply vivisect/__init__.py:1307]: We should. It'd probably be a breaking change unless it was a new function, but that would be good.

**Pull request merged**